internal/funcs/generated/gen/gen.go

```diff
@@ -0,0 +1,286 @@
+		return nil, err
+	}
+	defer i.Remove(ctx)
+
+	// log version
+	tf, err := tfexec.NewTerraform(installDir, execPath)
+	if err != nil {
+		return nil, err
+	}
+	coreVersion, _, err := tf.Version(ctx, true)
+	if err != nil {
+		return nil, err
+	}
+	log.Printf("using Terraform %s (%s)", coreVersion, execPath)
+
+	return tf.MetadataFunctions(ctx)
+}
+
+// signatureHash calculates the SHA256 checksum for the given function
+// signatures.
+func signatureHash(functions *tfjson.MetadataFunctions) (string, error) {
+	var rawJson bytes.Buffer
+	err := json.NewEncoder(&rawJson).Encode(functions)
+	if err != nil {
+		return "", fmt.Errorf("failed to encode functions: %w", err)
+	}
+
+	hash := sha256.Sum256(rawJson.Bytes())
+	return fmt.Sprintf("%x", hash), nil
+}
+
+// writeFunctions generates the Go code for the function signatures,
+// formats it and writes it to a new file.
+func writeFunctions(filename string, functions *tfjson.MetadataFunctions) {
+	outputTpl := `// Code generated by "gen"; DO NOT EDIT.
+package funcs
+
+import (
+	"github.com/hashicorp/hcl-lang/schema"
+	"github.com/zclconf/go-cty/cty"
+	"github.com/zclconf/go-cty/cty/function"
+)
+func {{ .FunctionName }}() map[string]schema.FunctionSignature {
+	return map[string]schema.FunctionSignature{
+{{- range $funcName, $func := .Signatures }}
+		{{- $varpar := $func.VariadicParameter }}
+		{{- $params := $func.Parameters }}
+		"{{ $funcName }}": {
+			{{- with $params }}
+			Params: []function.Parameter{
+				{{- range $key, $value := $params }}
+				{{- $desc := $value.Description }}
+				{
+					Name:        "{{ $value.Name }}",
+					{{- with $desc }}
+					Description: "{{ escapeDescription . }}",{{- end }}
+					Type:        {{ $value.Type.GoString }},
+				},
+				{{- end }}
+			},{{- end }}
+			{{- with $varpar }}
+			VarParam:    &function.Parameter{
+				Name:             "{{ .Name }}",
+				Description:      "{{ .Description }}",
+				Type:             {{ .Type.GoString }},
+			},{{- end }}
+			ReturnType: {{ $func.ReturnType.GoString }},
+			Description: "{{ escapeDescription $func.Description }}",
+		},	
+{{- end }}
+	}
+}
+`
+
+	tpl, err := template.New("output").Funcs(template.FuncMap{
+		"escapeDescription": escapeDescription,
+	}).Parse(outputTpl)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	type data struct {
+		FunctionName string
+		Signatures   map[string]*tfjson.FunctionSignature
+	}
+
+	var buf bytes.Buffer
+	err = tpl.Execute(&buf, data{
+		FunctionName: fmt.Sprintf("v%s_Functions", escapeVersion(terraformVersion)),
+		Signatures:   functions.Signatures,
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	p, err := format.Source(buf.Bytes())
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	f, err := os.OpenFile(filename, os.O_RDWR|os.O_CREATE, 0o644)
+	if err != nil {
+		log.Fatal(err)
+	}
+	_, err = f.Write(p)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func escapeDescription(description string) string {
+	return strings.ReplaceAll(description, `\`, `\\`)
```

Comment:
Again - not sure if the HTML escape characters come already in the JSON output - which would be something to address upstream, but either way we should make sure that double quotes are escaped correctly here too.